tests/feature_selection/test_significance_tests.py

```diff
@@ -0,0 +1,158 @@
+import pytest
+import numpy as np
+import pandas as pd
+
+from tsfresh.defaults import TEST_FOR_BINARY_TARGET_REAL_FEATURE
+from tsfresh.feature_selection.significance_tests import target_binary_feature_binary_test, \
+    target_binary_feature_real_test, target_real_feature_real_test, target_real_feature_binary_test
+
+
+@pytest.fixture()
+def binary_feature():
+    return pd.Series(np.random.binomial(1, 0.5, 250), name="TEST")
+
+
+@pytest.fixture()
+def binary_target_not_related():
+    return pd.Series(np.random.binomial(1, 0.5, 250))
+
+
+@pytest.fixture()
+def real_feature():
+    return pd.Series(np.random.normal(0, 1, 250), name="TEST")
+
+
+@pytest.fixture()
+def real_target_not_related():
+    return pd.Series(np.random.normal(0, 1, 250))
+
+
+class TestUnsignificant:
+    @pytest.fixture()
+    def minimal_p_value_for_unsignificant_features(self):
+        np.random.seed(seed=42)
```

Comment:
Well spotted! I naively thought that the fixtures are executed in the order they appear as arguments. However after googling a bit, I found out that's not the case. I'll fix it. The solution should be that the fixtures relying on the seed to be set, should depend on a fixture that does that, thus guaranteeing the order of execution.